comps/vectorstores/langchain/pathway/vectorstore_pathway.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+
+import pathway as pw
+from langchain import text_splitter
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceHubEmbeddings
+from pathway.xpacks.llm.parsers import ParseUnstructured
+from pathway.xpacks.llm.vector_store import VectorStoreServer
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(name)s %(levelname)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+# This creates a Pathway connector that tracks all the files in the `data/` directory.
+# Additions and modifications will be reflected on the index automatically.
+
+data = pw.io.fs.read(
```

Comment:
currently, the OPEA moved all data ingestion operation into the dataprep part and not more set a default file in the retriever. it would be better to follow the same style in https://github.com/opea-project/GenAIComps/tree/main/comps/retrievers/langchain/milvus to set the retriever and ingest the document in the dataprep part. Otherwise, there would be a default document that can impact the retrieval result

Reply:
I can remove the default PDF under the `data/` folder.

By dataprep, I suppose you mean something like [this](https://github.com/opea-project/GenAIComps/blob/main/comps/dataprep/milvus/prepare_doc_milvus.py).

The issue is, Pathway vector store takes the "input sources" during the construction, and keeps listening to these sources. Your PDFs and other files that are added to these folders are automatically indexed by Pathway (with the built-in parser, splitter, embedder). Hence, we don't have chunk `upsert` type of interface for the DB.
We explain this on our docs, I also tried to explain it in the `vectorstore_pathway.py`.

Do you think we could make an exception in this case?

A "work around" could be to have simple dataprep service that simply copies files into some folder Pathway is listening to, but, it is not a good solution since the Pathway input folders are configurable by the user.

Reply:
Hi @berkecanrizai,

If users want to add or delete a file from the Pathway vectorstore, they just need to modify the 'data/' folder, right? If that's the case, we should introduce an environment variable to represent the vector store path. The `dataprep` microservice should operate exclusively on this path, saving new files to it or deleting files from it.

Reply:
Yes, but that's not the whole picture. As you may see on the `pw.io.fs` line, users can add any input connectors `pw.io` as sources. Some connectors include local folders, Google Drive, SharePoint, etc. 

I have included the `data/` folder as an example in this case. 
Yes, a solution could be to have the `dataprep` microservice to save files to this `data/` folder. However, this may not be ideal since users can modify the `"sources"` config.
But surely this will work. 

In the end, if there is no option to exclude the Pathway from the other vector DBs in terms of the `dataprep` implementation, I can implement it as discussed above.

Reply:
We can merge it firstly as vector database part, and we should also consider `dataprep` part for the whole pipeline.